src/Parser/Main.ml

```diff
@@ -34,8 +34,8 @@ let rec repl_seq imported () =
     let def = make_nowhere (Lang.Surface.DReplExpr(Desugar.tr_expr e)) in
     Seq.Cons(def, repl_seq imported)
 
-  | Raw.REPL_Def def ->
-    Seq.Cons(Desugar.tr_def def, repl_seq imported)
+  | Raw.REPL_Defs defs ->
+      (List.fold_left (fun ac def () -> Seq.Cons(Desugar.tr_def def, ac)) (repl_seq imported) defs) ()
```

Comment:
Using `fold_left` for appending lists naturally gives us `rev_append` instead of `append`. The same problem occurs here: definitions are reversed during parsing. This can be exhibited by the following input to REPL.
```
> let x = ()
let y = x ;;
<stdin>:2:9: fatal error: Unbound variable x
>
```
On the other hand, the following input work, but it shouldn't.
```
> let y = x
let x = () ;;
> 
```

BTW, we try to keep source in good old 80-column style. This line is too long.